src/main/java/jpos/config/simple/xml/JavaxRegPopulator.java

```diff
@@ -0,0 +1,513 @@
+                        CurrentEntry.addProperty( attrs.getValue( "name" ), obj );
+                    }
+                    catch( Exception e )
+                    {
+                        CurrentEntry = null;
+                        String msg = "Invalid prop: name=" + attrs.getValue( "name" )
+                                + ":value=" + attrs.getValue( "value" );
+                        tracer.println( ": " + msg );
+                        lastLoadException =  new SAXException( msg, e );
+                    };
+                }
+            }
+            tracer.println( "" );
+        }
+
+        private void addOptionalProperty( String name, String value )
+        {
+            CurrentEntry.addProperty( name, value == null ? "" : value );
+        }
+
+        @Override
+        public void endElement( String uri, String lname, String qname )
+        {
+            if( TheException == null )
+                tracer.println( "EndElement: " + qname );
+            else
+            {
+                tracer.println( "EndElement: " + TheException.getMessage() );
+                TheException = null;
+            }
+            if( qname.equals( "JposEntry" ) )
+            {
+                if( CurrentEntry != null )
+                    jposEntryList.add( CurrentEntry );
+                CurrentEntry = null;
+                TheException = null;
+            }
+        }
+
+        //----------------------------------------------------------------
+        // ErrorHandler
+
+        @Override
+        public void error( SAXParseException e )
+        {
+            CurrentEntry = null;
+            TheException = e;
+        }
+
+        @Override
+        public void fatalError( SAXParseException e )
+        {
+            CurrentEntry = null;
+            TheException = e;
+        }
+
+        //----------------------------------------------------------------
+        // EntityResolver
+
+        @Override
+        public InputSource resolveEntity(String publicId, String systemId )
+        {
+            tracer.println( "JposEntityResolver:resolveEntity:publicId=" +
+                    publicId );
+
+            tracer.println( "JposEntityResolver:resolveEntity:systemId=" +
+                    systemId );
+
+            if( publicId.equals( DTD_DOC_TYPE_VALUE ) )
+            {
+                InputStream is =
+                        getClass().getResourceAsStream( DTD_FILE_NAME );
+
+                if ( is == null )
+                    is = findFileInClasspath( DTD_FILE_NAME );
+
+                if( is != null )
+                    return new InputSource( new InputStreamReader( is ) );
+            }
+
+            return null;
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Instance variables
+    //
+
+    protected String xmlFileName = DEFAULT_XML_FILE_NAME;
+
+    private List jposEntryList = new LinkedList();
```

Comment:
Type Safety: We should make it type safe as it is a local declaration, i.e.
`    private List<JposEntry> jposEntryList = new LinkedList<>();`

Reply:
Forgot one thing.
The original implementor used LinkedList quite often. 
However, we do not sort the list and do not insert. So, does not make sense to use the more sophisticated `LinkedList`. An `ArrayList`will do it too. 
```suggestion
    private List jposEntryList<JposEntry> = new ArrayList<>();
```